sdk/azidentity/managed_identity_credential_test.go

```diff
@@ -1,21 +1,190 @@
+	msiCred.client = newTestManagedIdentityClient(&ManagedIdentityCredentialOptions{HTTPClient: srv})
+	_, err = msiCred.GetToken(context.Background(), azcore.TokenRequestOptions{Scopes: []string{scope}})
+	if err != nil {
+		t.Fatalf("Received an error when attempting to retrieve a token")
+	}
+}
+
+func TestManagedIdentityCredential_GetTokenInAppServiceMockFail(t *testing.T) {
+	err := resetEnvironmentVarsForTest()
+	if err != nil {
+		t.Fatalf("Unable to set environment variables")
+	}
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithStatusCode(http.StatusUnauthorized))
+	testURL := srv.URL()
+	_ = os.Setenv("MSI_ENDPOINT", testURL.String())
+	_ = os.Setenv("MSI_SECRET", "secret")
+	msiCred := NewManagedIdentityCredential("", newDefaultManagedIdentityOptions())
+	msiCred.client = newTestManagedIdentityClient(&ManagedIdentityCredentialOptions{HTTPClient: srv})
+	_, err = msiCred.GetToken(context.Background(), azcore.TokenRequestOptions{Scopes: []string{scope}})
+	if err == nil {
+		t.Fatalf("Expected an error but did not receive one")
+	}
+}
+
+// func TestManagedIdentityCredential_GetTokenIMDSMock(t *testing.T) {
+// 	timeout := time.After(5 * time.Second)
+// 	done := make(chan bool)
+// 	go func() {
+// 		err := resetEnvironmentVarsForTest()
+// 		if err != nil {
+// 			t.Fatalf("Unable to set environment variables")
+// 		}
+// 		srv, close := mock.NewServer()
+// 		defer close()
+// 		srv.AppendResponse(mock.WithBody([]byte(`{"access_token": "ey0....", "expires_in": 3600}`)))
+// 		msiCred := NewManagedIdentityCredential("", newDefaultManagedIdentityOptions())
+// 		msiCred.client = newTestManagedIdentityClient(&ManagedIdentityCredentialOptions{HTTPClient: srv})
+// 		_, err = msiCred.GetToken(context.Background(), azcore.TokenRequestOptions{Scopes: []string{scope}})
+// 		if err == nil {
+// 			t.Fatalf("Cannot run IMDS test in this environment")
+// 		}
+// 		time.Sleep(550 * time.Millisecond)
+// 		done <- true
+// 	}()
+
+// 	select {
+// 	case <-timeout:
+// 		t.Fatal("Test didn't finish in time")
+// 	case <-done:
+// 	}
+// }
+
+func TestManagedIdentityCredential_GetTokenUnknownFail(t *testing.T) {
+	err := resetEnvironmentVarsForTest()
+	if err != nil {
+		t.Fatalf("Unable to set environment variables")
+	}
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithStatusCode(http.StatusUnauthorized))
+	// testURL := srv.URL()
+	_ = os.Setenv("MSI_ENDPOINT", "https://t .com")
+	msiCred := NewManagedIdentityCredential("", newDefaultManagedIdentityOptions())
+	msiCred.client = newTestManagedIdentityClient(&ManagedIdentityCredentialOptions{HTTPClient: srv})
+	_, err = msiCred.GetToken(context.Background(), azcore.TokenRequestOptions{Scopes: []string{scope}})
+	if err == nil {
+		t.Fatalf("Expected an error but did not receive one")
+	}
+}
+
+func newTestManagedIdentityClient(options *ManagedIdentityCredentialOptions) *managedIdentityClient {
```

Comment:
How is this different from the non-test version?

Reply:
I had an extra line of code there overriding the default HTTP transport, but since we added that field to ManagedIdentityCredentialOptions this isn't necessary anymore. I'm removing it. Thanks for catching that!